lib/grp.gi

```diff
@@ -4291,86 +4294,121 @@ end );
-      fi;
-      typ:=fam!.defaultFinitelyGeneratedGroupWithOneType;
-    else
-      if not IsBound(fam!.defaultGroupWithOneType) then
-        fam!.defaultGroupWithOneType:=
-          NewType(fam,IsGroup and IsAttributeStoringRep
-                      and HasGeneratorsOfMagmaWithInverses and HasOne);
+function( gens, id )
+local G,fam,typ,triv;
+
+  fam:=FamilyObj(gens);
+  triv:=fail; # might not find out
+
+  typ:=IsGroup and IsAttributeStoringRep 
+        and HasIsEmpty and HasGeneratorsOfMagmaWithInverses and HasOne;
+  if IsFinite(gens) then
+    typ:=typ and IsFinitelyGeneratedGroup;
+    if Length(gens)<=1 then
+      typ:=typ and HasIsCyclic and IsCyclic 
+          and HasIsCommutative and IsCommutative;
+    fi;
+
+    if Length(gens)>0 and CanEasilyCompareElements(gens) then
+      if ForAny(gens,x->x<>id) then
+        typ:=typ and HasIsTrivial and HasIsNonTrivial;
+        triv:=false;
+      else
+        typ:=typ and HasIsTrivial and HasIsNonTrivial;
+        triv:=true;
+        # basic consequence
+        typ:=typ and HasIsPerfectGroup and IsPerfectGroup;
       fi;
-      typ:=fam!.defaultGroupWithOneType;
     fi;
 
-    G:=rec();
+  fi;
+  typ:=NewType(fam,typ);
+
+  G:=rec();
+  if triv=fail then
     ObjectifyWithAttributes(G,typ,GeneratorsOfMagmaWithInverses,AsList(gens),
-                            One,id);
+                          IsEmpty,false,One,id);
+  else
+    ObjectifyWithAttributes(G,typ,GeneratorsOfMagmaWithInverses,AsList(gens),
+                          IsEmpty,false,One,id,
+                          IsTrivial,triv,IsNonTrivial,not triv);
+  fi;
 
-    return G;
+  return G;
 end );
 
-InstallMethod( GroupWithGenerators,
-    "method for empty list and element",
-    [ IsList and IsEmpty, IsMultiplicativeElementWithInverse ],
-    function( empty, id )
-    local G,fam,typ;
+InstallMethod( GroupWithGenerators,"method for empty list and element",
+  [ IsList and IsEmpty, IsMultiplicativeElementWithInverse ],
+  function( empty, id )
+local G,fam,typ;
 
-    fam:= CollectionsFamily( FamilyObj( id ) );
-    if not IsBound( fam!.defaultFinitelyGeneratedGroupWithOneType ) then
-      fam!.defaultFinitelyGeneratedGroupWithOneType:=
-        NewType( fam, IsGroup and IsAttributeStoringRep
-                      and HasGeneratorsOfMagmaWithInverses
-                      and IsFinitelyGeneratedGroup and HasOne );
-    fi;
-    typ:= fam!.defaultFinitelyGeneratedGroupWithOneType;
+  fam:= CollectionsFamily( FamilyObj( id ) );
 
-    G:= rec();
-    ObjectifyWithAttributes( G, typ,
-                             GeneratorsOfMagmaWithInverses, empty,
-                             One, id );
+  typ:=IsGroup and IsAttributeStoringRep 
+        and HasGeneratorsOfMagmaWithInverses and HasOne
+        and IsFinitelyGeneratedGroup and HasIsTrivial and HasIsNonTrivial
+        and HasIsFinite and HasIsCyclic
+        and HasIsPerfectGroup and HasIsEmpty;
```

Comment:
Can't you just replace the above with `typ:= IsGroup and IsAttributeStoringRep and HasGeneratorsOfMagmaWithInverses and HasOne and IsTrivial` ? The `IsTrivial` implies (via zero cost implications) all the remaining flags.